src/charts/donut/donut-pct-chart-directive.js

```diff
@@ -316,34 +347,25 @@ angular.module('patternfly.charts').directive('pfDonutPctChart', function (c3Cha
     link: function (scope, element) {
       var setupDonutChartTitle = function () {
         $timeout(function () {
-          var donutChartTitle, bigText, smText;
+          var donutChartTitle, centerLabelText;
 
           donutChartTitle = element[0].querySelector('text.c3-chart-arcs-title');
-          if (scope.config.centerLabelFn) {
-            donutChartTitle.innerHTML = scope.config.centerLabelFn(scope);
-          } else if (scope.centerLabel === 'none') {
-            donutChartTitle.innerHTML = '';
+          if (!donutChartTitle) {
+            return;
+          }
+
+          centerLabelText = scope.getCenterLabelText(scope);
+
+          if (scope.config.centerLabelFn || scope.centerLabel === 'none') {
+            donutChartTitle.innerHTML = centerLabelText.bigText;
```

Comment:
Ok, fixed the logic, made it easier to understand.  I'm ok with massaging the code in order to get more code coverage.